src/drivers/sdl/lv_sdl_window.c

```diff
@@ -208,71 +208,44 @@ static inline int sdl_render_mode(void)
-                dsc->rotated_buf = sdl_draw_buf_realloc_aligned(dsc->rotated_buf, buf_size);
-                dsc->rotated_buf_size = buf_size;
-            }
+        int32_t px_map_w = lv_area_get_width(area);
+        int32_t px_map_h = lv_area_get_height(area);
+        uint32_t px_map_stride = lv_draw_buf_width_to_stride(lv_area_get_width(area), cf);
+        uint32_t px_size = lv_color_format_get_size(cf);
 
-            switch(rotation) {
-                case LV_DISPLAY_ROTATION_0:
-                    break;
-                case LV_DISPLAY_ROTATION_90:
-                    lv_draw_sw_rotate(px_map, dsc->rotated_buf, w, h, w_stride, h_stride, rotation, cf);
-                    break;
-                case LV_DISPLAY_ROTATION_180:
-                    lv_draw_sw_rotate(px_map, dsc->rotated_buf, w, h, w_stride, w_stride, rotation, cf);
-                    break;
-                case LV_DISPLAY_ROTATION_270:
-                    lv_draw_sw_rotate(px_map, dsc->rotated_buf, w, h, w_stride, h_stride, rotation, cf);
-                    break;
-            }
+        int32_t fb_stride = disp->hor_res * px_size;
+        uint8_t * fb_start = dsc->fb_act;
+        fb_start += rotated_area.y1 * fb_stride + rotated_area.x1 * px_size;
+        lv_display_rotation_t rotation = lv_display_get_rotation(disp);
 
-            px_map = dsc->rotated_buf;
+        if(rotation == LV_DISPLAY_ROTATION_0) {
+            uint32_t px_map_line_bytes = lv_area_get_width(area) * px_size;
 
-            rotated_area = *area;
-            lv_display_rotate_area(disp, &rotated_area);
-            area = &rotated_area;
+            int32_t y;
+            for(y = area->y1; y <= area->y2; y++) {
+                lv_memcpy(fb_start, px_map, px_map_line_bytes);
+                px_map += px_map_stride;
+                fb_start += fb_stride;
+            }
         }
-
-        uint32_t px_map_stride = lv_draw_buf_width_to_stride(lv_area_get_width(area), cf);
-        uint32_t px_map_line_bytes = lv_area_get_width(area) * px_size;
-
-        uint8_t * fb_tmp = dsc->fb_act;
-        uint32_t fb_stride = disp->hor_res * px_size;
-        fb_tmp += area->y1 * fb_stride;
-        fb_tmp += area->x1 * px_size;
-
-        int32_t y;
-        for(y = area->y1; y <= area->y2; y++) {
-            lv_memcpy(fb_tmp, px_map, px_map_line_bytes);
-            px_map += px_map_stride;
-            fb_tmp += fb_stride;
+        else {
+            lv_draw_sw_rotate(px_map, fb_start, px_map_w, px_map_h, px_map_stride, fb_stride, rotation, cf);
         }
-    }
 
-    /* TYPICALLY YOU DO NOT NEED THIS
-     * If it was the last part to refresh update the texture of the window.*/
-    if(lv_display_flush_is_last(disp)) {
-        if(sdl_render_mode() != LV_DISPLAY_RENDER_MODE_PARTIAL) {
-            dsc->fb_act = px_map;
+        if(lv_display_flush_is_last(disp)) {
+            if(sdl_render_mode() != LV_DISPLAY_RENDER_MODE_PARTIAL) {
```

Comment:
This will never be true. This condition is in a block where `sdl_render_mode() == LV_DISPLAY_RENDER_MODE_PARTIAL` is true.

Reply:
Ah, thanks! It was off by one level.